### ISSUE TITLE
If a command line argument is a directory that contains a tox.ini file, that tox file is used

### ISSUE THREAD

**Issue by MattSable** [pytest-dev/pytest#4009]
If you supply a command line argument that is a directory that contains a tox.ini file, that tox file gets used. Example layout and files that can reproduce this:

```
/tests
    conftest.py
    test.py
/some_folder
    tox.ini
```

**conftest.py**
```
def pytest_addoption(parser):
    parser.addoption(
        "--foo"
    )
```

**test.py**
```
def test():
    print("Hello world!")
```

**tox.ini**
```
[pytest]
addopts = --collect-only
```

Running just `pytest .\tests\test.py`, I get the expected output:

```
============================= test session starts =============================
platform win32 -- Python 3.6.4, pytest-3.8.0, py-1.6.0, pluggy-0.7.1
rootdir: C:\Users\masable\Desktop\examples, inifile:
collected 1 item

tests\test.py .                                                          [100%]

========================== 1 passed in 0.03 seconds ===========================
```

If instead you run `pytest .\tests\test.py --foo .\some_folder\`, than the tox.ini in `some_folder` is used, and the output is:

```
============================= test session starts =============================
platform win32 -- Python 3.6.4, pytest-3.8.0, py-1.6.0, pluggy-0.7.1
rootdir: C:\Users\masable\Desktop\examples\some_folder, inifile: tox.ini
collected 1 item
<Module 'test.py'>
  <Function 'test'>

======================== no tests ran in 0.02 seconds =========================

```

I'm on pytest 3.8.0.

**Comment by nicoddemus** [comment]
Unfortunately that's a problem with the args parser: by the time it is figuring out the command-line, it doesn't know yet that `conftest.py` will create a `--foo` option, so it assumes `.\some_folder` is a directory to look for tests.

A workaround is to use `--foo=.\some_folder` instead.

Unfortunately this is not easy to fix because the parser at that time doesn't know yet about `--foo`. I think this might be related to #1149.

**Comment by Zac-HD** [comment]
I believe this is closed by #5469.
